### PR TITLE
#365 Making property names customizable

### PR DESCRIPTION
**Pull request by gunnarmorling**
Based on the work of @chschu this provides the naming strategy contract. I have simplified the contract a bit, moved the SPI package to the processor module and added an integration test. The latter is in a separate module for now, it may be added to the existing integrationtest module, but I couldn't make that work for the time being as the test needs two modules: one with the strategy, one which is using it.

### CODE REVIEW

File: parent/pom.xml [hunk 1]
```diff
@@ -207,6 +207,11 @@
                 <version>${project.version}</version>
             </dependency>
             <dependency>
+                <groupId>${project.groupId}</groupId>
+                <artifactId>mapstruct-spi</artifactId>
```
Comment (agudian) [inline parent/pom.xml]: that's not used anymore, right?

File: processor/src/main/java/org/mapstruct/ap/naming/package-info.java [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+/**
+ * <p>
+ * SPI (service provider interface) contracts to be implemented for advanced use cases.
```
Comment (agudian) [inline processor/src/main/java/org/mapstruct/ap/naming/package-info.java]: Is this the right doc for this package? As far as I can see, this package currently only holds the default implementation of this one SPI...?

File: processor/src/main/java/org/mapstruct/ap/util/Executables.java [hunk 3]
```diff
@@ -44,49 +43,30 @@
  */
 public class Executables {
 
+    private static AccessorNamingStrategy accessorNamingStrategy = Services.get(
```
Comment (agudian) [inline processor/src/main/java/org/mapstruct/ap/util/Executables.java]: Couldn't this also be `final`?

**Comment by agudian** [comment]
Looks good so far - I just have some small remarks.

If it's all the same for you, I would like to try to move the integration-test into the integration-test module (will do that later tonight)...

File: parent/pom.xml (hunk 1 above)
Comment (gunnarmorling) [inline parent/pom.xml]: True, that can be removed.

File: processor/src/main/java/org/mapstruct/ap/naming/package-info.java (hunk 2 above)
Comment (gunnarmorling) [inline processor/src/main/java/org/mapstruct/ap/naming/package-info.java]: Hum, yes, seems like I mixed those up.

File: processor/src/main/java/org/mapstruct/ap/util/Executables.java (hunk 3 above)
Comment (gunnarmorling) [inline processor/src/main/java/org/mapstruct/ap/util/Executables.java]: +1

**Comment by gunnarmorling** [comment]
> Looks good so far - I just have some small remarks.

Cool, thanks for the review.

> I would like to try to move the integration-test into the integration-test module

+1 That'd be my preference, I just struggled to do so :)

Would you like to apply those other changes you pointed out together with the move?

**Comment by agudian** [comment]
> Would you like to apply those other changes you pointed out together with the move?

Sure, I'll go over it later tonight.

Perhaps that helps you finding some time to.... you know... that other pr... :wink:

**Comment by gunnarmorling** [comment]
Yes, believe it or not, it's on my to-do list for tonight :) Thanks,
Andreas! Btw. there was quite some nice buzz on MapStruct on Twitter today
:)

**Comment by agudian** [comment]
I've worked in the comments, rebased the commits, moved the integration-test to the integration-test module (plain and simple, supporting only the `ORACLE_JAVA_8` processor type) and pushed to master.
@chschu, @gunnarmorling: thanks for this cool feature!

> Btw. there was quite some nice buzz on MapStruct on Twitter today :)
Great! :+1:

**Comment by chschu** [comment]
@gunnarmorling @agudian Thanks for your effort, this looks great now!

**Comment by gunnarmorling** [comment]
@chschu Thank you, too! Glad to hear you like the final outcome. Great to have this feature in place now.

**Pull request closed without merging**